src/modules/simulation/simulator_mavlink/SimulatorMavlink.cpp

```diff
@@ -1424,20 +1438,55 @@ void SimulatorMavlink::check_failure_injections()
 			handled = true;
 
 			if (failure_type == vehicle_command_s::FAILURE_TYPE_OFF) {
-				PX4_WARN("CMD_INJECT_FAILURE, mag off");
 				supported = true;
-				_mag_blocked = true;
+
+				// 0 to signal all
+				if (instance == 0) {
+					for (int i = 0; i < MAG_COUNT_MAX; i++) {
+						PX4_WARN("CMD_INJECT_FAILURE, mag %d off", i);
+						_mag_blocked[i] = true;
+						_mag_stuck[i] = false;
+					}
+
+				} else if (instance >= 1 && instance <= MAG_COUNT_MAX) {
+					PX4_WARN("CMD_INJECT_FAILURE, mag %d off", instance - 1);
+					_mag_blocked[instance - 1] = true;
+					_mag_stuck[instance - 1] = false;
+				}
```

Comment:
why is `MAG_COUNT_MAX` inclusive in the conditional and `instance - 1` used in the index? are the sensor instances 1-based?

Reply:
This is analogously implemented as for the gyros. Here you can set failures of the magnetometers and select the instance which should fails. If you select instance 0 this is equivalent as to set the failure on all magnetometers. with Instances 1 to MAG_COUNT_MAX you can set only the specific instance. This is only for the interface to set a failure.